extension/parquet/parquet_reader.cpp

```diff
@@ -355,13 +356,29 @@ unique_ptr<ColumnReader> ParquetReader::CreateReader(const duckdb_parquet::forma
 
 	auto &root_struct_reader = (StructColumnReader &)*ret;
 
-	// add casts if required
-	for (auto &entry : cast_map) {
-		auto column_idx = entry.first;
-		auto &expected_type = entry.second;
-		auto child_reader = move(root_struct_reader.child_readers[column_idx]);
-		auto cast_reader = make_unique<CastColumnReader>(move(child_reader), expected_type);
-		root_struct_reader.child_readers[column_idx] = move(cast_reader);
+	if (!parquet_options.union_by_name) {
```

Comment:
I'm not sure if modifying the Parquet Reader itself is necessary here - can't we handle the `union_by_name` in the `ParquetScanFunction` outside of the `ParquetReader` class? That should also allow for more code re-use between the CSV and Parquet readers for this functionality - by e.g. moving `SetNullUnionCols` to a shared `UnionByName` class.